src/funding/funding.ts

```diff
@@ -186,32 +199,37 @@ export class Funding {
+    await this.setAccountBalance();
+    //validate the deposit amount is not more than the account balance
     if (this.depositAmount.gt(this.accountBalance)) {
-      this.eventAggregator.publish("handleValidationError", new EventConfig(`The amount you wish to deposit (${BigNumber.from(this.depositAmount)} ${tokenSymbol}) exceeds the current balance in your account (${BigNumber.from(this.accountBalance)} ${tokenSymbol}). Please submit again.`, EventMessageType.Warning, "Insufficient Balance"));
+      this.eventAggregator.publish("handleValidationError", new EventConfig(`The amount you wish to deposit (${BigNumber.from(this.depositAmount)} ${depositToken.symbol}) exceeds the current balance in your account (${BigNumber.from(this.accountBalance)} ${depositToken.symbol}). Please submit again.`, EventMessageType.Warning, "Insufficient Balance"));
       this.depositAmount = this.accountBalance;
       return;
     }
-    if (this.depositAmount.gt(recentRequiredTokens)) {
-      this.eventAggregator.publish("handleValidationError", new EventConfig(`The amount you wish to deposit (${BigNumber.from(this.depositAmount)} ${tokenSymbol}) exceeds the required funding needed (${BigNumber.from(recentRequiredTokens)} ${tokenSymbol}). Please submit again.`, EventMessageType.Warning));
-      this.depositAmount = recentRequiredTokens;
+    //validate the deposit amount is not more than the required tokens to fund the deal
+    if (this.depositAmount.gt(depositToken.required)) {
+      this.eventAggregator.publish("handleValidationError", new EventConfig(`The amount you wish to deposit (${BigNumber.from(this.depositAmount)} ${depositToken.symbol}) exceeds the required funding needed (${BigNumber.from(depositToken.required)} ${depositToken.symbol}). Please submit again.`, EventMessageType.Warning));
+      this.depositAmount = depositToken.required;
       return;
     }
-    //TODO implement the deposit of tokens
-    this.eventAggregator.publish("handleInfo", new EventConfig(`Depositing ${BigNumber.from(this.depositAmount)} ${tokenSymbol} on behalf of ${this.firstDao.name}`, EventMessageType.Info, "Deposit Submitted"));
-    //TODO handle account provider transaction rejection
-    //TODO handle the popup notification on the event of the deposit actually being completed
+    //all validation passed so submit the deposit
+    const depositTransaction = await this.deal.depositTokens(this.firstDao, depositToken.address, this.depositAmount);
+    if (depositTransaction){
+      //deposit was successful
+      this.eventAggregator.publish("handleInfo", new EventConfig(`${BigNumber.from(this.depositAmount)} ${depositToken.symbol} has been deposited`, EventMessageType.Info, "Deposit completed"));
+    } else {
+      //something happened to make the deposit not happen
+      this.eventAggregator.publish("handleException", new EventConfigException("An error occurred while trying to deposit tokens. Please try again.", EventMessageType.Exception));
```

Comment:
best to use "handleFailure" and just pass it a string for the message.

Reply:
Done